sklearn/metrics/classification.py

```diff
@@ -528,6 +528,7 @@ def matthews_corrcoef(y_true, y_pred, sample_weight=None):
     y_pred = lb.transform(y_pred)
 
     C = confusion_matrix(y_true, y_pred, sample_weight=sample_weight)
+    C = C.astype(np.float64)    # Bug#9622
     t_sum = C.sum(axis=1)
     p_sum = C.sum(axis=0)
     n_correct = np.trace(C)
```

Comment:
can we use `n_samples = len(y_true)` to avoid imprecision there?

Reply:
no, because of `sample_weight`

Reply:
good point. Thanks